lib/dfe/analytics/shared/checksum_logic.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+# Logic shared between entity_table_check and import_entity_table_check
+module ChecksumLogic
+  TIME_ZONE = 'London'
+
+  def fetch_current_timestamp_in_time_zone
+    result = ActiveRecord::Base.connection.select_all('SELECT CURRENT_TIMESTAMP AS current_timestamp')
+    result.first['current_timestamp'].in_time_zone(TIME_ZONE).iso8601(6)
+  end
+
+  def supported_adapter_and_environment?
+    return true if adapter_name == 'postgresql' || !Rails.env.production?
+
+    Rails.logger.info('DfE::Analytics: Entity checksum: Only Postgres databases supported on PRODUCTION')
+
+    false
+  end
+
+  def entity_table_check_data(entity, order_column)
+    checksum_calculated_at = fetch_current_timestamp_in_time_zone
+
+    row_count, checksum = fetch_checksum_data(entity, checksum_calculated_at, order_column)
+    Rails.logger.info("DfE::Analytics Processing entity: #{entity}: Row count: #{row_count}")
+    {
+      row_count: row_count,
+      checksum: checksum,
+      checksum_calculated_at: checksum_calculated_at,
+      order_column: order_column
+    }
+  end
+
+  def determine_order_column(entity, columns)
+    if ActiveRecord::Base.connection.column_exists?(entity, :updated_at) && columns.include?('updated_at')
+      'UPDATED_AT'
+    elsif ActiveRecord::Base.connection.column_exists?(entity, :created_at) && columns.include?('created_at')
+      'CREATED_AT'
+    else
```

Comment:
We need to reintroduce the order by "ID" column.

Reply:
Oh yes, I'd forgotten about that!